src/AstTransforms.cpp

```diff
@@ -1292,4 +1292,39 @@ bool PolymorphicOperatorsTransformer::transform(AstTranslationUnit& translationU
     return update.changed;
 }
 
+bool AstPragmaChecker::transform(AstTranslationUnit& translationUnit) {
```

Comment:
I think it would be better to have an AstPragmaChecker.h/cpp than to put all the definitions in AstTransforms.cpp/h

I had actually already done this in https://github.com/souffle-lang/souffle/compare/master...mmcgr:moveTransformers , along with the other transformers hidden around the codebase, so that shows what I mean. I was about to make a PR of the changes, but I can cut out the AstPragmaChecker changes and submit it later.

Reply:
I am easy on this; I just want to disentangle the AstNode classes from everything else. I feel we should have for each transformer an own header/cpp file as well.

Reply:
What do you suggest for the time being?

Reply:
I'll close this in favour of my pull request, since it seems to contain the same changes as you were after, plus the other transformers.